{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[]},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":[],"metadata":{"id":"kWNubH1OAeMh"}},{"cell_type":"markdown","source":["Rodar no ambiente conda já criado bioinfo2"],"metadata":{"id":"6hed4P3pAgWh"}},{"cell_type":"code","execution_count":null,"metadata":{"id":"QRbdEFSBAZ51"},"outputs":[],"source":["# baixando o arquivo de trabalho\n","wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr22.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz"]},{"cell_type":"code","source":["# baixando o metadado\n","wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/integrated_call_samples_v3.20130502.ALL.panel"],"metadata":{"id":"1TuFBTO_BA9A"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# baixando o arquivo tabix\n","wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr22.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz.tbi"],"metadata":{"id":"0fKGJoTOBUKH"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# Selecionar 30 amostras aleatórias (15 casos + 15 controles)\n","shuf integrated_call_samples_v3.20130502.ALL.panel | head -n 30 | awk '{print $1}' > samples.txt\n","\n","# Criar arquivo de fenótipos\n","awk 'NR<=15 {print $1, $1, 2} NR>15 {print $1, $1, 1}' samples.txt > pheno.txt"],"metadata":{"id":"Si2WGgXZBxr3"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# filtrar vcf MAF > 0.05% para dar poder estatistico\n","bcftools view -S samples.txt -i 'MAF[0] > 0.0005' -Oz -o chr22_filtered.vcf.gz ALL.chr22.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz"],"metadata":{"id":"a21XWMePCCZ3"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["###codigo adicional para gerar o arquivo 'sample.txt' apartir do arquivo 'pheno.txt'\n","cut -d' ' -f1 dia_pheno.txt > dia_pheno_clean.txt\n"],"metadata":{"id":"Zm-eFg9QelhJ"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# Converter para PLINK e Filtros Adicionais\n","plink --vcf chr22_filtered.vcf.gz --make-bed --maf 0.05 --geno 0.05 --hwe 1e-6 --allow-extra-chr --out chr22_clean\n","plink --vcf ad_wxs_15filtered.vcf.gz --make-bed --maf 0.05 --geno 0.05 --hwe 1e-6 --allow-extra-chr --out ad_wxs_clean\n","plink --vcf dia_wxs_15filtered.vcf.gz --make-bed --double-id --maf 0.05 --geno 0.05 --hwe 1e-6 --allow-extra-chr --out dia_wxs_clean"],"metadata":{"id":"Xrkqn42cCcfF"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# rodar o GWAS\n","plink --bfile chr22_clean --pheno pheno.txt --assoc --allow-no-sex --out gwas_results\n","plink --bfile ad_wxs_clean --pheno ad_pheno.txt --assoc --allow-no-sex --out gwas_ad_results\n","plink --bfile dia_wxs_clean --pheno dia_pheno.txt --assoc --allow-no-sex --out gwas_dia_results"],"metadata":{"id":"_SYgSiw3Cvha","colab":{"base_uri":"https://localhost:8080/","height":108},"outputId":"d405a2cd-9f0c-48a0-f9d0-923cd3629796"},"execution_count":null,"outputs":[{"output_type":"error","ename":"SyntaxError","evalue":"invalid syntax (<ipython-input-1-e5e51f4c96b0>, line 2)","traceback":["\u001b[0;36m  File \u001b[0;32m\"<ipython-input-1-e5e51f4c96b0>\"\u001b[0;36m, line \u001b[0;32m2\u001b[0m\n\u001b[0;31m    plink --bfile chr22_clean --pheno pheno.txt --assoc --allow-no-sex --out gwas_results\u001b[0m\n\u001b[0m                  ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"]}]},{"cell_type":"code","source":["# conferindo os resultados:\n","#1. Verifique as primeiras linhas do arquivo:\n","head -n 5 gwas_results.assoc\n","#2. Verifique os nomes das colunas:\n","awk 'NR==1 {print; exit}' gwas_results.assoc\n","#3. Conte o número de SNPs:\n","wc -l gwas_results.assoc  # Total de linhas\n","awk 'NR>1' gwas_results.assoc | wc -l  # SNPs apenas (excluindo cabeçalho)\n","#4. Verifique a distribuição dos valores-p:\n","# Para valores-p não missing:\n","awk '$9 != \"NA\" && NR>1 {print $9}' gwas_results.assoc | sort -g | uniq -c | head\n","# Verifique valores problematicos\n","# Valores não numéricos na coluna P (geralmente coluna 9):\n","awk 'NR>1 && $9 !~ /^[0-9.eE-]+$/ {print $9}' gwas_results.assoc | sort | uniq -c"],"metadata":{"id":"KbiyNt0CEfqI"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# gerando script em R para gerar os graficos:\n","nano generate_plots_fixed.R"],"metadata":{"id":"UttIaS2XFY0t"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["#!/usr/bin/env Rscript\n","\n","# Instalar pacote se necessário\n","if (!require(\"qqman\")) {\n","    install.packages(\"qqman\", repos=\"https://cloud.r-project.org\")\n","    library(qqman)\n","}\n","\n","# Ler dados\n","gwas_data <- read.table(\"gwas_dia_results.assoc\", header=TRUE)\n","\n","# Criar IDs únicos para SNPs (já que a coluna SNP tem \".\")\n","gwas_data$SNP_ID <- paste0(\"chr\", gwas_data$CHR, \":\", gwas_data$BP)\n","\n","# Verificar e limpar valores-p\n","gwas_data$P <- as.numeric(gwas_data$P)\n","valid_data <- gwas_data[!is.na(gwas_data$P) & is.finite(gwas_data$P), ]\n","\n","# 1. Manhattan Plot\n","png(\"manhattan_dia_final.png\", width=1200, height=600)\n","manhattan(valid_data,\n","          chr = \"CHR\",\n","          bp = \"BP\",\n","          p = \"P\",\n","          snp = \"SNP_ID\",  # Usar IDs criados\n","          main = \"Manhattan Plot - GWAS Results\",\n","          suggestiveline = -log10(1e-5),\n","          genomewideline = -log10(5e-8))\n","dev.off()\n","\n","# 2. QQ Plot\n","png(\"qq_dia_final.png\", width=600, height=600)\n","qq(valid_data$P, main = \"QQ Plot - GWAS Results\")\n","dev.off()\n","\n","print(\"Gráficos gerados com sucesso! Verifique: manhattan_final.png e qq_final.png\")"],"metadata":{"id":"ECW4Hwf0GLLW","colab":{"base_uri":"https://localhost:8080/","height":108},"outputId":"b17a5daf-15df-42bd-b279-3274a6b655fd"},"execution_count":null,"outputs":[{"output_type":"error","ename":"SyntaxError","evalue":"invalid syntax (<ipython-input-3-e4928181f3b4>, line 4)","traceback":["\u001b[0;36m  File \u001b[0;32m\"<ipython-input-3-e4928181f3b4>\"\u001b[0;36m, line \u001b[0;32m4\u001b[0m\n\u001b[0;31m    if (!require(\"qqman\")) {\u001b[0m\n\u001b[0m        ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"]}]},{"cell_type":"code","source":["Rscript generate_plots_fixed.R"],"metadata":{"id":"yFuAMjVvGOMp"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# 1. SNPs com p < 0.05 (significativos)\n","awk 'NR==1 || $9 < 0.05' gwas_results.assoc > snps_significativos.txt\n","awk 'NR==1 || $9 < 0.05' gwas_ad_results.assoc > snps_ad_significativos.txt\n","awk 'NR==1 || $9 < 0.05' gwas_dia_results.assoc > snps_dia_significativos.txt\n","\n","# 2. SNPs com significância genômica (p < 5e-8)\n","awk 'NR==1 || $9 < 5e-8' gwas_results.assoc > snps_genomicos.txt"],"metadata":{"id":"oDf2z0zcQ_Dw"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["# Apartir daqui prepararemos para analise de pleiotropia"],"metadata":{"id":"L_RXIBVESUw5"}},{"cell_type":"code","source":["# Passo 1: Executar o GWAS para ambas as doenças\n","# Doença 1 (exemplo)\n","plink --bfile chr22_filtered \\\n","      --pheno pheno_doenca1.txt \\\n","      --assoc \\\n","      --out gwas_doenca1\n","\n","# Doença 2\n","plink --bfile chr22_filtered \\\n","      --pheno pheno_doenca2.txt \\\n","      --assoc \\\n","      --out gwas_doenca2"],"metadata":{"id":"6r3OpKSISaxU"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# Passo 2: Identificar SNPs significativos em ambas as doenças\n","# SNPs com p < 0.05 em ambas (ajuste o limiar conforme necessário)\n","awk 'NR==FNR && $9 < 0.05 {print $2; next}\n","     NR!=FNR && $9 < 0.05 && ($2 in snps) {print $2}' \\\n","     gwas_ad_results.assoc gwas_dia_results.assoc > snps_pleiotropicos.txt\n"],"metadata":{"id":"A73zZliMSh8Z"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["#Passo 3: Análise estatística de pleiotropia\n","#Opção 1: Teste de heterogeneidade (Cochran's Q)\n","# Combinar resultados (requer colunas: SNP, P_DOENCA1, P_DOENCA2)\n","paste <(awk 'NR>1 {print $2, $9}' gwas_ad_results.assoc) \\\n","      <(awk 'NR>1 {print $9}' gwas_dia_results.assoc) > combined_pvalues.txt\n","# remover linhas que não tenham 3 colunas\n","# Criar versão filtrada (mantém apenas linhas com 3 colunas)\n","awk 'NF==3' combined_pvalues.txt > combined_pvalues_filtered.txt\n","\n","# Rodar diretamente no terminal\n","Rscript -e '\n","  data <- read.table(\"combined_pvalues_filtered.txt\", col.names=c(\"SNP\", \"P1\", \"P2\"))\n","  data$Q <- -2*(log(data$P1) + log(data$P2))  # Estatística Q\n","  data$P_pleio <- pchisq(data$Q, df=2, lower.tail=FALSE)\n","  write.table(data, \"pleiotropy_results.txt\", quote=F, row.names=F)'"],"metadata":{"id":"BThHyppISpH8"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["#Opção 2: Correlação de efeitos (β)\n","# Extrair efeitos (OR ou β)\n","paste <(awk 'NR>1 {print $2, $10}' gwas_ad_results.assoc) \\\n","      <(awk 'NR>1 {print $10}' gwas_dia_results.assoc) > combined_effects.txt\n","\n","# Rodar diretamente no terminal\n","\n","awk 'NF==3' combined_effects.txt > combined_effects_clean.txt\n","\n","Rscript -e '\n","  data <- read.table(\"combined_effects_clean.txt\", col.names=c(\"SNP\", \"OR1\", \"OR2\"))\n","  cor_test <- cor.test(data$OR1, data$OR2, method=\"spearman\")\n","  print(paste(\"Correlação de efeitos (rho):\", cor_test$estimate))'\n","\n","  # resultado após rodar esse script : Warning message:\n","# In cor.test.default(data$OR1, data$OR2, method = \"spearman\") :\n","#  Cannot compute exact p-value with ties\n","#[1] \"Correlação de efeitos (rho): 0.00226277031660643\"\n"],"metadata":{"id":"zQNRKutXSx1x"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# Passo 4: Visualização\n","# Scatter plot de efeitos pleiotrópicos (R)\n","Rscript -e '\n","  # Carregar dados do arquivo corrigido\n","  data <- read.table(\"combined_effects_clean.txt\", col.names=c(\"SNP\", \"OR1\", \"OR2\"))\n","\n","  # Passo 4: Visualização\n","  # Scatter plot de efeitos pleiotrópicos\n","  png(\"pleiotropy_scatter.png\")\n","  plot(data$OR1, data$OR2,\n","       xlab=\"OR Alzheimer\", ylab=\"OR Diabetes type2\",\n","       main=\"Efeitos Pleiotrópicos\")\n","  abline(h=1, v=1, lty=2, col=\"red\")\n","  dev.off()\n","'"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":108},"id":"NguBllaYS2zO","outputId":"3d4c913a-57b8-427b-9f3a-8863680de910"},"execution_count":null,"outputs":[{"output_type":"error","ename":"SyntaxError","evalue":"invalid syntax (<ipython-input-4-12c14c07c875>, line 4)","traceback":["\u001b[0;36m  File \u001b[0;32m\"<ipython-input-4-12c14c07c875>\"\u001b[0;36m, line \u001b[0;32m4\u001b[0m\n\u001b[0;31m    plot(data$OR1, data$OR2,\u001b[0m\n\u001b[0m             ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"]}]},{"cell_type":"code","source":["# Manhattan plot combinado\n","Rscript -e '\n","  # Carregar pacote qqman\n","  library(qqman)\n","\n","  # Carregar dados GWAS\n","  gwas_alzheimer <- read.table(\"gwas_ad_results.assoc\", header=TRUE)\n","  gwas_diabetes_type2 <- read.table(\"gwas_dia_results.assoc\", header=TRUE)\n","\n","  # Carregar SNPs pleiotrópicos\n","  snps_pleiotropicos <- read.table(\"snps_pleiotropicos.txt\", header=FALSE)$V1\n","\n","  # Criar PDF com dois Manhattan plots\n","  pdf(\"manhattan_plots_combined.pdf\", width=12, height=10)\n","  par(mfrow=c(2,1))\n","\n","  # Manhattan plot para Alzheimer\n","  manhattan(gwas_alzheimer, highlight=snps_pleiotropicos, col=c(\"blue4\", \"skyblue\"),\n","           main=\"Manhattan Plot - Alzheimer\")\n","\n","  # Manhattan plot para Diabetes tipo 2\n","  manhattan(gwas_diabetes_type2, highlight=snps_pleiotropicos, col=c(\"red4\", \"indianred1\"),\n","           main=\"Manhattan Plot - Type 2 diabetes\")\n","\n","  dev.off()\n","\n","  # Informar ao usuário\n","  cat(\"Plots gerados no arquivo manhattan_plots_combined.pdf\\n\")\n","'"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":108},"id":"TrjWW2PCTBjk","outputId":"aa79cdf9-ed29-4709-edd6-d0a66edc2a67"},"execution_count":null,"outputs":[{"output_type":"error","ename":"SyntaxError","evalue":"invalid syntax (<ipython-input-5-ebe84cfb921a>, line 2)","traceback":["\u001b[0;36m  File \u001b[0;32m\"<ipython-input-5-ebe84cfb921a>\"\u001b[0;36m, line \u001b[0;32m2\u001b[0m\n\u001b[0;31m    qqman::manhattan(gwas_alzheimer, highlight=snps_pleiotropicos, col=\"blue\") +\u001b[0m\n\u001b[0m          ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"]}]},{"cell_type":"code","source":["# Passo 5: Anotação funcional (opcional)\n","# Usar ANNOVAR ou VEP para anotar SNPs pleiotrópicos\n","annotate_variants.pl snps_pleiotropicos.txt -out pleio_annotated"],"metadata":{"id":"yHGcSLwhTGdr"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":[" Dicas importantes\n","\n","    Controle de FDR: Aplique correção para múltiplos testes (ex: Benjamini-Hochberg) nos resultados pleiotrópicos.\n","\n","    Meta-análise: Se as doenças forem correlacionadas, use métodos como MTAG.\n","\n","    Banco de dados: Consulte o *GWAS Catalog* para ver se seus SNPs já foram reportados como pleiotrópicos.\n","\n","\n","Observação:\n","FID  IID  PHENO\n","ID1  ID1  2     # Caso (doente)\n","ID2  ID2  1     # Controle (saudável)\n","ID3  ID3  -9    # Missing (opcional)\n","\n","outras opçoes:\n","\n","FID  IID  PHENO  IDADE  SEXO\n","ID1  ID1  2      45     1\n","ID2  ID2  1      30     2"],"metadata":{"id":"I2UXGx3cTQcA"}},{"cell_type":"markdown","source":["Analisando o arquivo pleiotropy_results.txt que você obteve, posso explicar o significado das colunas:\n","Coluna 1: ID do SNP (por exemplo, \"78196:G\")\n","Coluna 2 (P1): P-valor da associação com Alzheimer (do arquivo gwas_ad_results.assoc)\n","Coluna 3 (P2): P-valor da associação com Diabetes tipo 2 (do arquivo gwas_dia_results.assoc)\n","Coluna 4 (Q): Estatística Q de Fisher combinada (-2*(log(P1) + log(P2)))\n","Coluna 5 (P_pleio): P-valor do teste de pleiotropia (significância da estatística Q)\n","Com esses dados, você pode criar vários tipos de gráficos informativos:\n","\n","Gráfico de Manhattan para pleiotropia:\n","\n"],"metadata":{"id":"cGpEhqWguP8u"}},{"cell_type":"code","source":["# Gráfico de Manhattan para pleiotropia:\n","Rscript -e '\n","  data <- read.table(\"pleiotropy_results.txt\", header=TRUE)\n","  # Extrair cromossomo e posição do ID do SNP\n","  data$CHR <- as.numeric(sapply(strsplit(as.character(data$SNP), \":\"), function(x) x[1]))\n","  data$BP <- as.numeric(sapply(strsplit(as.character(data$SNP), \":\"), function(x) x[2]))\n","  # Criar Manhattan plot\n","  library(qqman)\n","  png(\"pleiotropy_manhattan.png\", width=1000, height=600, res=100)\n","  manhattan(data, chr=\"CHR\", bp=\"BP\", p=\"P_pleio\",\n","            main=\"Manhattan Plot - Efeitos Pleiotrópicos\",\n","            suggestiveline=-log10(0.05), genomewideline=-log10(0.001))\n","  dev.off()\n","'"],"metadata":{"id":"vwveptyiuaOc"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# Gráfico de Dispersão dos p-valores:\n","Rscript -e '\n","  data <- read.table(\"pleiotropy_results.txt\", header=TRUE)\n","  png(\"pvalue_scatter.png\", width=800, height=800, res=100)\n","  plot(-log10(data$P1), -log10(data$P2),\n","       xlab=\"-log10(P-valor Alzheimer)\", ylab=\"-log10(P-valor Type2 Diabetes)\",\n","       main=\"Significance Comparison between Phenotypes\",\n","       pch=19, col=ifelse(data$P_pleio < 0.05, \"red\", \"black\"))\n","  abline(h=-log10(0.05), v=-log10(0.05), lty=2, col=\"blue\")\n","  legend(\"topright\", legend=c(\"Pleiotropy (p<0.05)\", \"Not significant\"),\n","         col=c(\"red\", \"black\"), pch=19)\n","  dev.off()\n","'"],"metadata":{"id":"82RwDcd1uf-e"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["\n","# Volcano plot para pleiotropia:\n","Rscript -e '\n","  data <- read.table(\"pleiotropy_results.txt\", header=TRUE)\n","  png(\"pleiotropy_volcano.png\", width=800, height=800, res=100)\n","  plot(data$Q, -log10(data$P_pleio),\n","       xlab=\"Estatística Q\", ylab=\"-log10(P-valor pleiotropia)\",\n","       main=\"Volcano Plot - Pleiotropia\",\n","       pch=19, col=ifelse(data$P_pleio < 0.05, \"red\", \"black\"))\n","  abline(h=-log10(0.05), lty=2, col=\"blue\")\n","  legend(\"topright\", legend=c(\"Pleiotropia (p<0.05)\", \"Não significativo\"),\n","         col=c(\"red\", \"black\"), pch=19)\n","  dev.off()\n","'"],"metadata":{"id":"UMeAbneGutGO"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["mais graficos:"],"metadata":{"id":"FBHYG5buz6Td"}},{"cell_type":"code","source":["Rscript -e '\n","  data <- read.table(\"pleiotropy_results.txt\", header=TRUE)\n","  png(\"pleiotropy_pval_histogram.png\", width=800, height=600, res=100)\n","\n","  # Criar histograma dos p-valores\n","  hist(data$P_pleio,\n","       breaks = 20,  # Número de barras\n","       xlab = \"P-valor de pleiotropia\",\n","       ylab = \"Frequência\",\n","       main = \"Distribuição dos P-valores de Pleiotropia\",\n","       col = \"skyblue\",\n","       border = \"white\")\n","\n","  # Adicionar linha de significância\n","  abline(v = 0.05, col = \"red\", lty = 2, lwd = 2)\n","  legend(\"topright\", legend = \"Limite de significância (p < 0.05)\",\n","         col = \"red\", lty = 2, lwd = 2)\n","\n","  dev.off()\n","'"],"metadata":{"id":"DLLQS1Wsz8kd"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["Rscript -e '\n","  data <- read.table(\"pleiotropy_results.txt\", header=TRUE)\n","  png(\"pleiotropy_pval_histogram.png\", width=800, height=600, res=100)\n","\n","  # Separar p-valores significativos e não significativos\n","  sig <- data$P_pleio < 0.05\n","\n","  hist(data$P_pleio[!sig],\n","       breaks = 20,\n","       xlab = \"P-valor de pleiotropia\",\n","       ylab = \"Frequência\",\n","       main = \"Distribuição dos P-valores de Pleiotropia\",\n","       col = \"gray\",\n","       border = \"white\")\n","\n","  hist(data$P_pleio[sig],\n","       breaks = 20,\n","       col = \"red\",\n","       border = \"white\",\n","       add = TRUE)\n","\n","  abline(v = 0.05, col = \"blue\", lty = 2, lwd = 2)\n","  legend(\"topright\",\n","         legend = c(\"P < 0.05\", \"P ≥ 0.05\"),\n","         fill = c(\"red\", \"gray\"))\n","\n","  dev.off()\n","'"],"metadata":{"id":"oO1RY5JB0G5E"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["Rscript -e '\n","  data <- read.table(\"pleiotropy_results.txt\", header=TRUE)\n","  png(\"pleiotropy_qqplot.png\", width=800, height=800, res=100)\n","\n","  qqplot(-log10(ppoints(length(data$P_pleio))),\n","         -log10(sort(data$P_pleio)),\n","         xlab = \"Valores esperados (-log10 P)\",\n","         ylab = \"Valores observados (-log10 P)\",\n","         main = \"QQ-plot dos P-valores de Pleiotropia\")\n","  abline(0, 1, col = \"red\")\n","\n","  dev.off()\n","'"],"metadata":{"id":"pyNy4jo90ICV"},"execution_count":null,"outputs":[]}]}